File: leetcode/best-time-to-buy-and-sell-stock/solution_test.cpp
#include "../../tools/leetcode.h"
#include "solution.cpp"

int main() {
    Solution sol;
    vector<int> prices1{7, 1, 5, 3, 6, 4};
    vector<int> prices2{7, 6, 4, 3, 1};
    vector<int> prices3{1};
    test(sol.maxProfit(prices1) == 5, "Example 1");
    test(sol.maxProfit(prices2) == 0, "Example 2");
    test(sol.maxProfit(prices3) == 0, "Single price");
    return summary();
}

File: leetcode/best-time-to-buy-and-sell-stock/solution_test.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", ".."))
from tools.leetcode import test, summary
from solution import Solution

sol = Solution()
test(sol.maxProfit([7, 1, 5, 3, 6, 4]), 5, "Example 1")
test(sol.maxProfit([7, 6, 4, 3, 1]), 0, "Example 2")
test(sol.maxProfit([1]), 0, "Single price")

summary()
